tracing-journald/src/memfd.rs

```diff
@@ -0,0 +1,35 @@
+//! memfd helpers.
+
+use libc::*;
+use std::fs::File;
+use std::io::Error;
+use std::io::Result;
+use std::os::raw::c_uint;
+use std::os::unix::prelude::{FromRawFd, RawFd};
+
+fn create(flags: c_uint) -> Result<File> {
+    unsafe {
+        let fd = memfd_create("tracing-journald\0".as_ptr() as *const c_char, flags);
+        if fd < 0 {
+            Err(Error::last_os_error())
+        } else {
+            Ok(File::from_raw_fd(fd as RawFd))
+        }
```

Comment:
we could limit the `unsafe` block here to just the actual unsafe operation:
```suggestion
    let fd = unsafe { memfd_create("tracing-journald\0".as_ptr() as *const c_char, flags) };
    if fd < 0 {
        Err(Error::last_os_error())
    } else {
        Ok(File::from_raw_fd(fd as RawFd))
    }
```